app/models/nic/base.rb

```diff
@@ -144,16 +150,30 @@ def host_managed?
       self.host && self.host.managed? && SETTINGS[:unattended]
     end
 
-    def require_ip_validation?
+    def require_ip_validation?(ip_field, other_ip_field, subnet)
+      raise "invalid ip field #{ip_field}" unless ip_field == :ip || ip_field == :ip6
+      raise "invalid other ip field #{other_ip_field}" unless other_ip_field == :ip || other_ip_field == :ip6
```

Comment:
Please use

    raise Foreman::Exception, '...'

instead which will generate an "ERF" code (http://projects.theforeman.org/projects/foreman/wiki/ErrorCodes). If this is user facing bug, then it should be also translated (http://projects.theforeman.org/projects/foreman/wiki/Translating).